service/orchestrator/assessment_results_test.go

```diff
@@ -51,6 +51,77 @@ import (
 	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
+func TestService_GetAssessmentResult(t *testing.T) {
+	type fields struct {
+		storage persistence.Storage
+	}
+	type args struct {
+		ctx context.Context
+		req *orchestrator.AssessmentResultRequest
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		req     *orchestrator.AssessmentResultRequest
+		res     *assessment.AssessmentResult
+		wantErr assert.ErrorAssertionFunc
+	}{
+		{
+			name: "request is missing",
+			fields: fields{
+				storage: testutil.NewInMemoryStorage(t),
+			},
+			args:    args{},
+			res:     nil,
+			wantErr: assert.Error,
```

Comment:
When you use `assert.Error`, you just know that SOME error occurred. But for the case of existing errors, you want to have checks that is really the error you are targeting (here it can be 3 different kinds of error).
Thus, it is better to create a function here and check the respective errors.
See example of last error case (internal) in the other comment